tizen-api/include/tizen-api.h

```diff
@@ -0,0 +1,300 @@
+ * @param[in] handle The nns_src_handle returned by nns_pipeline_gethandle().
+ * @param[out] policy The policy of buf deallocation.
+ * @param[in] buf The input buffer, in the format of tensorinfo given by nns_pipeline_gethandle()
+ * @param[in] size The size of input buffer. This must be consistent with the given tensorinfo.
+ * @return 0 on success (buf is filled). otherwise a negative error value.
+ * @retval #NNS_ERROR_NONE Successful
+ */
+int nns_pipeline_src_inputdata (nns_src_handle *handle,
+    nns_buf_policy policy, char *buf, size_t size)
+
+/****************************************************
+ ** NNStreamer Pipeline Switch/Valve Control       **
+ ****************************************************/
+
+/**
+ * @brief Get a handle to operate a "GstInputSelector / GstOutputSelector" node of nnstreamer pipelines
+ * @detail Refer to gstreamer.freedesktop.org/data/doc/gstreamer/head/gst-plugins-bad-plugins/html/gst-plugins-bad-plugins-input-selector.html for input selectors
+ *         Refer to https://gstreamer.freedesktop.org/data/doc/gstreamer/head/gstreamer-plugins/html/gstreamer-plugins-output-selector.html for output selectors
+ * @param[in] pipe The pipeline to be managed.
+ * @param[in] switchname The name of switch (InputSelector/OutputSelector)
+ * @param[out] num_nodes The number of nodes selected by this switch node. We assume they are listed from 0 ... num_nodes-1.
+ * @param[in] type The type of the switch.
+ * @return The switch handle. Null if error.
+ */
+nns_switch_handle nns_pipeline_switch_gethandle
+(nns_pipeline_h *pipe, const char *switchname, int *num_nodes, nns_switch_type type);
+
+/**
+ * @brief Close the given switch handle
+ * @param[in] h The handle to be closed
+ * @return @c 0 on success. otherwise a negative error value
+ * @retval #NNS_ERROR_NONE Successful
+ */
+int nns_pipeline_switch_close (nns_switch_handle h);
+
+/**
+ * @brief Control the switch with the given handle
+ * @param[in] h The switch handle returned by nns_pipeline_switch_gethandle()
+ * @param[in] node The node number, which is 0 ... num_nodes-1.
+ * @return @c 0 on success. otherwise a negative error value
+ * @retval #NNS_ERROR_NONE Successful
+ */
+int nns_pipeline_switch_control (nns_switch_handle h, int node);
+
+/**
+ * @brief Get a handle to operate a "GstValve" node of nnstreamer pipelines
+ * @detail Refer to https://gstreamer.freedesktop.org/data/doc/gstreamer/head/gstreamer-plugins/html/gstreamer-plugins-valve.html for more info.
+ * @param[in] pipe The pipeline to be managed.
+ * @param[in] valvename The name of valve (Valve)
+ * @return The valve handle. Null if error.
+ */
+nns_valve_handle nns_pipeline_valve_gethandle
+(nns_pipeline_h *pipe, const char *valvename);
+
+/**
+ * @brief Close the given valve handle
+ * @param[in] h The handle to be closed
+ * @return @c 0 on success. otherwise a negative error value
+ * @retval #NNS_ERROR_NONE Successful
+ */
+int nns_pipeline_valve_close (nns_valve_handle h);
```

Comment:
rename function to `nns_pipeline_valve_close_handle` to remove confusion with closing the valve usinng `nns_pipeline_valve_control (h, 0)`

Reply:
Renamed: ```nns_pipeline_valve_puthandle ```

Reply:
Thanks